### PR TITLE
Get building again with ASGARD_USE_MATLAB=ON

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/project-asgard/asgard/wiki/developing)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Changes required to successfully build with ASGARD_USE_MATLAB=ON

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [x] Bugfix
- [ ] New feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

## What systems has this change been tested on?

my laptop

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- [x] this PR is up to date with current the current state of 'develop'
- [x] code added or changed in the PR has been clang-formatted
- [ ] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

### CODE REVIEW

File: src/matlab_plot.cpp [hunk 1]
```diff
@@ -209,7 +209,7 @@ fk::vector<T> matlab_plot::generate_nodes(int const degree, int const level,
 
   // TODO: fully implement the output_grid options from matlab (this is just
   // the 'else' case)
-  T const roots =
+  auto const roots =
```
Comment (mkstoyanov) [inline src/matlab_plot.cpp]: I don't like the overuse of `auto`, if a type is simple, e.g., `int` `double` `T`, then we should just use the type.

`auto` is useful when the output depends on template parameters or is something potentially messy (e.g., some crazy iterator).

File: src/time_advance.cpp [hunk 2]
```diff
@@ -571,6 +571,9 @@ imex_advance(PDE<P> &pde, matrix_list<P> &operator_matrices,
 
   auto do_poisson_update = [&](fk::vector<P, mem_type::owner, imex_resrc> const
                                    &f_in) {
+    fk::vector<P> poisson_source(quad_dense_size);
+    fk::vector<P> phi(quad_dense_size);
+    fk::vector<P> poisson_E(quad_dense_size);
```
Comment (mkstoyanov) [inline src/time_advance.cpp:576]: I guess `tools::time_event` wasn't such a good idea after all ...

**Review by mkstoyanov — CHANGES_REQUESTED** [review]
Just change the `auto`, the rest of the changes contain lots of **ouch**, good you caught those.

File: src/matlab_plot.cpp (hunk 1 above)
Comment (mkstoyanov) [reply src/matlab_plot.cpp]: If the type is `fk::vector<T>` then how did it work before with just `T`?

Comment (quantumsteve) [reply src/matlab_plot.cpp]: I tried searching the history yesterday but didn't figure that out. Line 215 calls `roots.size()` which wouldn't work for T = float or T = double.

File: src/time_advance.cpp (hunk 2 above)
Comment (quantumsteve) [reply src/time_advance.cpp:576]: It shows the potential benefits of a CI runner with MATLAB or removing the MATLAB code entirely.

File: src/matlab_plot.cpp (hunk 1 above)
Comment (mkstoyanov) [reply src/matlab_plot.cpp]: Very strange ... `legendre_weights` does return two `fk::vectors` so the `fk::vector` should be correct.

File: src/time_advance.cpp (hunk 2 above)
Comment (mkstoyanov) [reply src/time_advance.cpp:576]: We are going  towards no-MATLAB but we have to make sure we don't have a major regression in plotting if we were to remove the code. Should check with Stefan today.

Either way the JOSS submission can include MATLAB for now.

**Review by mkstoyanov — APPROVED** [review]

**Pull request merged**